map/framework.cpp

```diff
@@ -1109,43 +1076,12 @@ void Framework::SetCurrentCountryChangedListener(TCurrentCountryChanged const &
 
 void Framework::UpdateUserViewportChanged()
 {
-  if (IsInteractiveSearchActive())
+  if (IsViewportSearchActive())
   {
-    (void)GetCurrentPosition(m_lastInteractiveSearchParams.m_lat,
-                             m_lastInteractiveSearchParams.m_lon);
-    Search(m_lastInteractiveSearchParams);
-  }
-}
-
-bool Framework::Search(search::SearchParams const & params)
-{
-#ifdef FIXED_LOCATION
-  search::SearchParams rParams(params);
-  if (params.IsValidPosition())
-  {
-    m_fixedPos.GetLat(rParams.m_lat);
-    m_fixedPos.GetLon(rParams.m_lon);
+    auto & intent = m_searchIntents[static_cast<size_t>(search::Mode::Viewport)];
```

Comment:
Because you only need to get params, it's a little bit shorter:
```cpp
auto const & params = m_searchIntents[static_cast<size_t>(search::Mode::Viewport)].m_params;
```